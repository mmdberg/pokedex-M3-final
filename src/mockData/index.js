export const mockPokemonTypeData = [
  {
    "id": "1",
    "name": "normal",
    "pokemon": [
      "16",
      "17",
      "18",
      "19",
      "20"
    ]
  },
]