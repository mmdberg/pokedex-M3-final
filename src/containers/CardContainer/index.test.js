import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps } from './index';
import React from 'react';
import { mockPokemonTypeData } from '../../mockData';

describe('CardContainer', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<CardContainer typeData={ mockPokemonTypeData }/>)
    expect(wrapper).toMatchSnapshot()
  });

  describe('mapStateToProps', () => {
    it('should correctly map typeData to props', () => {
      const mockState = {
        typeData: mockPokemonTypeData
      }
      const mapped = mapStateToProps(mockState)
      expect(mapped.typeData).toEqual(mockPokemonTypeData)
    })
  })
})