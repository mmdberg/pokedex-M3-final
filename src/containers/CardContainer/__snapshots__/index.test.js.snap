// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    typeData={
      Array [
        Object {
          "id": "1",
          "name": "normal",
          "pokemon": Array [
            "16",
            "17",
            "18",
            "19",
            "20",
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          className="card"
          handleClick={[Function]}
          pokemon={Array []}
          type={
            Object {
              "id": "1",
              "name": "normal",
              "pokemon": Array [
                "16",
                "17",
                "18",
                "19",
                "20",
              ],
            }
          }
        />,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "className": "card",
          "handleClick": [Function],
          "pokemon": Array [],
          "type": Object {
            "id": "1",
            "name": "normal",
            "pokemon": Array [
              "16",
              "17",
              "18",
              "19",
              "20",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            className="card"
            handleClick={[Function]}
            pokemon={Array []}
            type={
              Object {
                "id": "1",
                "name": "normal",
                "pokemon": Array [
                  "16",
                  "17",
                  "18",
                  "19",
                  "20",
                ],
              }
            }
          />,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "className": "card",
            "handleClick": [Function],
            "pokemon": Array [],
            "type": Object {
              "id": "1",
              "name": "normal",
              "pokemon": Array [
                "16",
                "17",
                "18",
                "19",
                "20",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
